### ISSUE TITLE
Question about the transformation from grid coordinates to image coordinates?

### ISSUE THREAD

**Issue by rockywind** [TRAILab/CaDDN#51]
Thanks for your help on the previous issue.
The depth map is downsampling 4x, but the intrinsic is the same as the original one.
I think the transformation is below that.
![image](https://user-images.githubusercontent.com/14838781/124486467-35cd6c80-dde0-11eb-8a96-df77cd20837b.png)

**Comment by codyreading** [comment]
When performing the transformation, we utilize normalized coordinates, which are essentially coordinates normalized between [-1, 1]. That way, the transformation works regardless of scale.

What ends up happening exactly is we generate a set of points in the center of each voxel, which we call the sampling grid. We project each point in the sampling grid from 3D space into the camera frustum space of full size (not downsampled by 4). Then, we normalize each point between [-1, 1], forming our frustum sampling grid. We then use the [grid_sample](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html#torch.nn.functional.grid_sample) function to sample from our frustum grid, which accepts normalized coordinates. Since coordinates are normalized between [-1, 1], the functionality will be the same regardless of scale of image features/depth map.

**Comment by rockywind** [comment]
Thanks a lot! @codyreading
